src/encoded/search.py

```diff
@@ -819,43 +868,310 @@ def set_filters(request, search, result, principals, doc_types):
+    :param options: list of options for that field
+
+    Example: field_name='embedded.files.file_size.raw', options=[20, 30, 40]
+
+    :return: dsl-subquery that is effectively an OR of all options on the field. See SHOULD.
+    """
+    should_query = {BOOL: {SHOULD: {TERMS: {field_name: []}}}}
+    for option in options:
+        should_query[BOOL][SHOULD][TERMS][field_name].append(option)
+    return should_query
+
+
+def handle_nested_filters(nested_filters, final_filters, key='must'):
+    """ Helper function for set_filters
+    Collapses nested filters together into a single query
+    ** Modifies final_filters and final_filters in place **
+
+    :param nested_filters: All nested fields that we would like to search on
+    :param final_filters: Collection of filters formatted in lucene, to be extended with nested filters
+    :param key: 'must' or 'must_not'
+    """
+    KEY_MAP = {MUST: MUST_NOT, MUST_NOT: MUST}
+    if key not in KEY_MAP:
+        raise RuntimeError('Tried to handle nested filter with key other than must/must_not: %s' % key)
+
+    # iterate through all nested filters
+    for field, query in nested_filters:
+
+        # iterate through all sub_query parts - note that this is modified in place hence the need
+        # to re-iterate after every nested filer is applied
+        nested_path = extract_nested_path_from_field(field)
+        sub_queries = final_filters['bool'][key]
+        found = False
+        for _q in sub_queries:
+
+            # Try to add to an existing 'nested' sub-query if possible
+            if _q.get(NESTED, None):
+                if _q[NESTED][PATH] == nested_path:
+                    try:
+                        # check if this field has multiple options
+                        options = query[BOOL][key][0][MATCH][field].split(',')
+                        if len(options) > 1:
+
+                            # construct SHOULD sub-query for all options
+                            sub_query = handle_should_query(field, options)
+                            _q[NESTED][QUERY][BOOL][key].append(sub_query)
+
+                        # if we don't have options, our original 'query' is what we need
+                        else:
+                            _q[NESTED][QUERY][BOOL][key].append(query[BOOL][key][0])
+
+                        found = True  # break is not sufficient, see below
+                        break
+                    except:       # Why? We found a 'range' nested query and must add this one separately
+                        continue  # This behavior is absurd. Somehow it knows to combine separate nested range
+                                  # queries with AND, but of course not regular queries and of course you cannot
+                                  # combine the range query here due to syntax  - Will
+        if not found:
+
+            # It's possible we're looking at a sub-query that's wrapped in a (length 1) list
+            if type(query) == list:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Using type() instead of isinstance() for a typecheck.](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=5350270)